app/code/Magento/InventoryConfigurableProduct/view/adminhtml/web/js/components/sources-visibility-checker.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'uiRegistry',
+    'uiComponent'
+], function (registry, component) {
+    'use strict';
+
+    return component.extend({
+        defaults: {
+            sourcesIndex: ''
+        },
+
+        /**
+         * Hide source tab if convert product to configurable and show it if to simple.
+         */
+        applySourcesConfiguration: function (visibleMatrix) {
+            registry.get('index = ' + this.sourcesIndex).visible(!visibleMatrix);
```

Comment:
Looks like we will have JS error if element will be not found (`get('index = ' + this.sourcesIndex)`)

Reply:
I've added check for undefined case.